### ISSUE TITLE
Refactor encoding/json with easyjson

### ISSUE THREAD

**Issue by Ja7ad** [meilisearch/meilisearch-go#579]
**Description**  
We propose replacing the built-in `encoding/json` package with `easyjson` to enhance performance for marshalling and unmarshalling large datasets. The built-in package is reliable but lacks performance optimizations for high-throughput scenarios. Refactoring to `easyjson` would provide significant speed improvements. However, there are limitations with `interface{}` types as encountered in #436, so careful handling of dynamic types will be necessary.

**Basic example**  
```go
// Replace json.Marshal with easyjson.Marshal
easyjson.Marshal(data)
```

**Comment by Ja7ad** [comment]
Hits field with type `*[]map[string]json.RawMessage` and unmarshaled value fixed without write custom marshaler.

![Screenshot from 2024-10-02 15-41-55](https://github.com/user-attachments/assets/dcc03c39-f4b7-4d08-9484-d9232dfe0f58)
![Screenshot from 2024-10-02 15-42-12](https://github.com/user-attachments/assets/5f199077-d491-44b8-a3f8-b3195bc1fd41)
![Screenshot from 2024-10-02 15-42-25](https://github.com/user-attachments/assets/39245395-b3e4-415e-a64e-a79e211a47f9)

**Comment by Ja7ad** [comment]
`easyjson` make complex client for unmarshaling.

**Comment by Ja7ad** [comment]
> Is this issue still open @Ja7ad . Wanted to pick it up

Yes I'm working on this issue https://github.com/meilisearch/meilisearch-go/pull/582

**Comment by curquiza** [comment]
@hegdeshashank73 thank you for your interest. The issue is already picked up, however, you are always welcome to provide review on the PR #582 😊
